components/menu/index.tsx

```diff
@@ -325,6 +330,7 @@ class InternalMenu extends React.Component<InternalMenuProps, MenuState> {
         value={{
           inlineCollapsed: this.getInlineCollapsed() || false,
           antdMenuTheme: this.props.theme,
+          direction: this.direction,
```

Comment:
Is that possible to switch `MenuContext.Provider` with `ConfigConsumer` to let `direction` from `ConfigConsumer` to the `MenuContext.Provider`? Since use `this. direction` may failed in concurrent mode.